// @gobii-ai/client/src/models/AgentList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface AgentList
 */
export interface AgentList {
    /**
     * 
     * @type {string}
     * @memberof AgentList
     */
    readonly id?: string;
    /**
     * 
     * @type {string}
     * @memberof AgentList
     */
    name: string;
    /**
     * 
     * @type {Date}
     * @memberof AgentList
     */
    readonly createdAt?: Date;
}

/**
 * Check if a given object implements the AgentList interface.
 */
export function instanceOfAgentList(value: object): value is AgentList {
    if (!('name' in value) || value['name'] === undefined) return false;
    return true;
}

export function AgentListFromJSON(json: any): AgentList {
    return AgentListFromJSONTyped(json, false);
}

export function AgentListFromJSONTyped(json: any, ignoreDiscriminator: boolean): AgentList {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'] == null ? undefined : json['id'],
        'name': json['name'],
        'createdAt': json['created_at'] == null ? undefined : (new Date(json['created_at'])),
    };
}

export function AgentListToJSON(json: any): AgentList {
    return AgentListToJSONTyped(json, false);
}

export function AgentListToJSONTyped(value?: Omit<AgentList, 'id'|'created_at'> | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'name': value['name'],
    };
}
