// @gobii-ai/client/src/models/AgentDetail.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface AgentDetail
 */
export interface AgentDetail {
    /**
     * 
     * @type {string}
     * @memberof AgentDetail
     */
    readonly id?: string;
    /**
     * 
     * @type {string}
     * @memberof AgentDetail
     */
    readonly userEmail?: string;
    /**
     * 
     * @type {string}
     * @memberof AgentDetail
     */
    name: string;
    /**
     * 
     * @type {Date}
     * @memberof AgentDetail
     */
    readonly createdAt?: Date;
    /**
     * 
     * @type {Date}
     * @memberof AgentDetail
     */
    readonly updatedAt?: Date;
}

/**
 * Check if a given object implements the AgentDetail interface.
 */
export function instanceOfAgentDetail(value: object): value is AgentDetail {
    if (!('name' in value) || value['name'] === undefined) return false;
    return true;
}

export function AgentDetailFromJSON(json: any): AgentDetail {
    return AgentDetailFromJSONTyped(json, false);
}

export function AgentDetailFromJSONTyped(json: any, ignoreDiscriminator: boolean): AgentDetail {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'] == null ? undefined : json['id'],
        'userEmail': json['user_email'] == null ? undefined : json['user_email'],
        'name': json['name'],
        'createdAt': json['created_at'] == null ? undefined : (new Date(json['created_at'])),
        'updatedAt': json['updated_at'] == null ? undefined : (new Date(json['updated_at'])),
    };
}

export function AgentDetailToJSON(json: any): AgentDetail {
    return AgentDetailToJSONTyped(json, false);
}

export function AgentDetailToJSONTyped(value?: Omit<AgentDetail, 'id'|'user_email'|'created_at'|'updated_at'> | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'name': value['name'],
    };
}
