// @gobii-ai/client/src/models/CancelTaskResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface CancelTaskResponse
 */
export interface CancelTaskResponse {
    /**
     * 
     * @type {string}
     * @memberof CancelTaskResponse
     */
    status: string;
    /**
     * 
     * @type {string}
     * @memberof CancelTaskResponse
     */
    message: string;
}

/**
 * Check if a given object implements the CancelTaskResponse interface.
 */
export function instanceOfCancelTaskResponse(value: object): value is CancelTaskResponse {
    if (!('status' in value) || value['status'] === undefined) return false;
    if (!('message' in value) || value['message'] === undefined) return false;
    return true;
}

export function CancelTaskResponseFromJSON(json: any): CancelTaskResponse {
    return CancelTaskResponseFromJSONTyped(json, false);
}

export function CancelTaskResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): CancelTaskResponse {
    if (json == null) {
        return json;
    }
    return {
        
        'status': json['status'],
        'message': json['message'],
    };
}

export function CancelTaskResponseToJSON(json: any): CancelTaskResponse {
    return CancelTaskResponseToJSONTyped(json, false);
}

export function CancelTaskResponseToJSONTyped(value?: CancelTaskResponse | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'status': value['status'],
        'message': value['message'],
    };
}
