// @gobii-ai/client/src/models/CancelTaskConflictResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface CancelTaskConflictResponse
 */
export interface CancelTaskConflictResponse {
    /**
     * 
     * @type {string}
     * @memberof CancelTaskConflictResponse
     */
    detail: string;
}

/**
 * Check if a given object implements the CancelTaskConflictResponse interface.
 */
export function instanceOfCancelTaskConflictResponse(value: object): value is CancelTaskConflictResponse {
    if (!('detail' in value) || value['detail'] === undefined) return false;
    return true;
}

export function CancelTaskConflictResponseFromJSON(json: any): CancelTaskConflictResponse {
    return CancelTaskConflictResponseFromJSONTyped(json, false);
}

export function CancelTaskConflictResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): CancelTaskConflictResponse {
    if (json == null) {
        return json;
    }
    return {
        
        'detail': json['detail'],
    };
}

export function CancelTaskConflictResponseToJSON(json: any): CancelTaskConflictResponse {
    return CancelTaskConflictResponseToJSONTyped(json, false);
}

export function CancelTaskConflictResponseToJSONTyped(value?: CancelTaskConflictResponse | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'detail': value['detail'],
    };
}
