// @gobii-ai/client/src/models/PaginatedAgentListList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { AgentList } from './AgentList';
import {
    AgentListFromJSON,
    AgentListFromJSONTyped,
    AgentListToJSON,
    AgentListToJSONTyped,
} from './AgentList';

/**
 * 
 * @export
 * @interface PaginatedAgentListList
 */
export interface PaginatedAgentListList {
    /**
     * 
     * @type {number}
     * @memberof PaginatedAgentListList
     */
    count: number;
    /**
     * 
     * @type {string}
     * @memberof PaginatedAgentListList
     */
    next?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PaginatedAgentListList
     */
    previous?: string | null;
    /**
     * 
     * @type {Array<AgentList>}
     * @memberof PaginatedAgentListList
     */
    results: Array<AgentList>;
}

/**
 * Check if a given object implements the PaginatedAgentListList interface.
 */
export function instanceOfPaginatedAgentListList(value: object): value is PaginatedAgentListList {
    if (!('count' in value) || value['count'] === undefined) return false;
    if (!('results' in value) || value['results'] === undefined) return false;
    return true;
}

export function PaginatedAgentListListFromJSON(json: any): PaginatedAgentListList {
    return PaginatedAgentListListFromJSONTyped(json, false);
}

export function PaginatedAgentListListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAgentListList {
    if (json == null) {
        return json;
    }
    return {
        
        'count': json['count'],
        'next': json['next'] == null ? undefined : json['next'],
        'previous': json['previous'] == null ? undefined : json['previous'],
        'results': ((json['results'] as Array<any>).map(AgentListFromJSON)),
    };
}

export function PaginatedAgentListListToJSON(json: any): PaginatedAgentListList {
    return PaginatedAgentListListToJSONTyped(json, false);
}

export function PaginatedAgentListListToJSONTyped(value?: PaginatedAgentListList | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'count': value['count'],
        'next': value['next'],
        'previous': value['previous'],
        'results': ((value['results'] as Array<any>).map(AgentListToJSON)),
    };
}
