// @gobii-ai/client/src/models/PaginatedTaskListList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { TaskList } from './TaskList';
import {
    TaskListFromJSON,
    TaskListFromJSONTyped,
    TaskListToJSON,
    TaskListToJSONTyped,
} from './TaskList';

/**
 * 
 * @export
 * @interface PaginatedTaskListList
 */
export interface PaginatedTaskListList {
    /**
     * 
     * @type {number}
     * @memberof PaginatedTaskListList
     */
    count: number;
    /**
     * 
     * @type {string}
     * @memberof PaginatedTaskListList
     */
    next?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PaginatedTaskListList
     */
    previous?: string | null;
    /**
     * 
     * @type {Array<TaskList>}
     * @memberof PaginatedTaskListList
     */
    results: Array<TaskList>;
}

/**
 * Check if a given object implements the PaginatedTaskListList interface.
 */
export function instanceOfPaginatedTaskListList(value: object): value is PaginatedTaskListList {
    if (!('count' in value) || value['count'] === undefined) return false;
    if (!('results' in value) || value['results'] === undefined) return false;
    return true;
}

export function PaginatedTaskListListFromJSON(json: any): PaginatedTaskListList {
    return PaginatedTaskListListFromJSONTyped(json, false);
}

export function PaginatedTaskListListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedTaskListList {
    if (json == null) {
        return json;
    }
    return {
        
        'count': json['count'],
        'next': json['next'] == null ? undefined : json['next'],
        'previous': json['previous'] == null ? undefined : json['previous'],
        'results': ((json['results'] as Array<any>).map(TaskListFromJSON)),
    };
}

export function PaginatedTaskListListToJSON(json: any): PaginatedTaskListList {
    return PaginatedTaskListListToJSONTyped(json, false);
}

export function PaginatedTaskListListToJSONTyped(value?: PaginatedTaskListList | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'count': value['count'],
        'next': value['next'],
        'previous': value['previous'],
        'results': ((value['results'] as Array<any>).map(TaskListToJSON)),
    };
}
