// @gobii-ai/client/src/models/PatchedAgentDetail.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface PatchedAgentDetail
 */
export interface PatchedAgentDetail {
    /**
     * 
     * @type {string}
     * @memberof PatchedAgentDetail
     */
    readonly id?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAgentDetail
     */
    readonly userEmail?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAgentDetail
     */
    name?: string;
    /**
     * 
     * @type {Date}
     * @memberof PatchedAgentDetail
     */
    readonly createdAt?: Date;
    /**
     * 
     * @type {Date}
     * @memberof PatchedAgentDetail
     */
    readonly updatedAt?: Date;
}

/**
 * Check if a given object implements the PatchedAgentDetail interface.
 */
export function instanceOfPatchedAgentDetail(value: object): value is PatchedAgentDetail {
    return true;
}

export function PatchedAgentDetailFromJSON(json: any): PatchedAgentDetail {
    return PatchedAgentDetailFromJSONTyped(json, false);
}

export function PatchedAgentDetailFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedAgentDetail {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'] == null ? undefined : json['id'],
        'userEmail': json['user_email'] == null ? undefined : json['user_email'],
        'name': json['name'] == null ? undefined : json['name'],
        'createdAt': json['created_at'] == null ? undefined : (new Date(json['created_at'])),
        'updatedAt': json['updated_at'] == null ? undefined : (new Date(json['updated_at'])),
    };
}

export function PatchedAgentDetailToJSON(json: any): PatchedAgentDetail {
    return PatchedAgentDetailToJSONTyped(json, false);
}

export function PatchedAgentDetailToJSONTyped(value?: Omit<PatchedAgentDetail, 'id'|'user_email'|'created_at'|'updated_at'> | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'name': value['name'],
    };
}
