// @gobii-ai/client/src/models/TaskList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { StatusEnum } from './StatusEnum';
import {
    StatusEnumFromJSON,
    StatusEnumFromJSONTyped,
    StatusEnumToJSON,
    StatusEnumToJSONTyped,
} from './StatusEnum';

/**
 * 
 * @export
 * @interface TaskList
 */
export interface TaskList {
    /**
     * 
     * @type {string}
     * @memberof TaskList
     */
    readonly id?: string;
    /**
     * 
     * @type {string}
     * @memberof TaskList
     */
    readonly agentId?: string;
    /**
     * 
     * @type {string}
     * @memberof TaskList
     */
    readonly prompt?: string | null;
    /**
     * Optional JSON schema to define structured output from the agent
     * @type {any}
     * @memberof TaskList
     */
    readonly outputSchema?: any | null;
    /**
     * 
     * @type {StatusEnum}
     * @memberof TaskList
     */
    readonly status?: StatusEnum;
    /**
     * 
     * @type {Date}
     * @memberof TaskList
     */
    readonly createdAt?: Date;
    /**
     * 
     * @type {Date}
     * @memberof TaskList
     */
    readonly updatedAt?: Date;
}



/**
 * Check if a given object implements the TaskList interface.
 */
export function instanceOfTaskList(value: object): value is TaskList {
    return true;
}

export function TaskListFromJSON(json: any): TaskList {
    return TaskListFromJSONTyped(json, false);
}

export function TaskListFromJSONTyped(json: any, ignoreDiscriminator: boolean): TaskList {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'] == null ? undefined : json['id'],
        'agentId': json['agent_id'] == null ? undefined : json['agent_id'],
        'prompt': json['prompt'] == null ? undefined : json['prompt'],
        'outputSchema': json['output_schema'] == null ? undefined : json['output_schema'],
        'status': json['status'] == null ? undefined : StatusEnumFromJSON(json['status']),
        'createdAt': json['created_at'] == null ? undefined : (new Date(json['created_at'])),
        'updatedAt': json['updated_at'] == null ? undefined : (new Date(json['updated_at'])),
    };
}

export function TaskListToJSON(json: any): TaskList {
    return TaskListToJSONTyped(json, false);
}

export function TaskListToJSONTyped(value?: Omit<TaskList, 'id'|'agent_id'|'prompt'|'output_schema'|'status'|'created_at'|'updated_at'> | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
    };
}
