// @gobii-ai/client/src/models/TaskDetail.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { StatusEnum } from './StatusEnum';
import {
    StatusEnumFromJSON,
    StatusEnumFromJSONTyped,
    StatusEnumToJSON,
    StatusEnumToJSONTyped,
} from './StatusEnum';

/**
 * 
 * @export
 * @interface TaskDetail
 */
export interface TaskDetail {
    /**
     * 
     * @type {string}
     * @memberof TaskDetail
     */
    readonly id?: string;
    /**
     * 
     * @type {string}
     * @memberof TaskDetail
     */
    agent?: string | null;
    /**
     * 
     * @type {string}
     * @memberof TaskDetail
     */
    readonly agentId?: string;
    /**
     * 
     * @type {string}
     * @memberof TaskDetail
     */
    prompt?: string | null;
    /**
     * Optional JSON schema to define structured output from the agent
     * @type {any}
     * @memberof TaskDetail
     */
    outputSchema?: any | null;
    /**
     * 
     * @type {StatusEnum}
     * @memberof TaskDetail
     */
    readonly status?: StatusEnum;
    /**
     * 
     * @type {Date}
     * @memberof TaskDetail
     */
    readonly createdAt?: Date;
    /**
     * 
     * @type {Date}
     * @memberof TaskDetail
     */
    readonly updatedAt?: Date;
    /**
     * 
     * @type {string}
     * @memberof TaskDetail
     */
    readonly errorMessage?: string | null;
    /**
     * 
     * @type {number}
     * @memberof TaskDetail
     */
    wait?: number;
    /**
     * Optional secrets for the task. Keys will be available as placeholders in the prompt.
     * @type {{ [key: string]: string; }}
     * @memberof TaskDetail
     */
    secrets?: { [key: string]: string; };
}



/**
 * Check if a given object implements the TaskDetail interface.
 */
export function instanceOfTaskDetail(value: object): value is TaskDetail {
    return true;
}

export function TaskDetailFromJSON(json: any): TaskDetail {
    return TaskDetailFromJSONTyped(json, false);
}

export function TaskDetailFromJSONTyped(json: any, ignoreDiscriminator: boolean): TaskDetail {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'] == null ? undefined : json['id'],
        'agent': json['agent'] == null ? undefined : json['agent'],
        'agentId': json['agent_id'] == null ? undefined : json['agent_id'],
        'prompt': json['prompt'] == null ? undefined : json['prompt'],
        'outputSchema': json['output_schema'] == null ? undefined : json['output_schema'],
        'status': json['status'] == null ? undefined : StatusEnumFromJSON(json['status']),
        'createdAt': json['created_at'] == null ? undefined : (new Date(json['created_at'])),
        'updatedAt': json['updated_at'] == null ? undefined : (new Date(json['updated_at'])),
        'errorMessage': json['error_message'] == null ? undefined : json['error_message'],
        'wait': json['wait'] == null ? undefined : json['wait'],
        'secrets': json['secrets'] == null ? undefined : json['secrets'],
    };
}

export function TaskDetailToJSON(json: any): TaskDetail {
    return TaskDetailToJSONTyped(json, false);
}

export function TaskDetailToJSONTyped(value?: Omit<TaskDetail, 'id'|'agent_id'|'status'|'created_at'|'updated_at'|'error_message'> | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {
        
        'agent': value['agent'],
        'prompt': value['prompt'],
        'output_schema': value['outputSchema'],
        'wait': value['wait'],
        'secrets': value['secrets'],
    };
}
