// @gobii-ai/client/src/apis/AgentsApi.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


import * as runtime from '../runtime';
import type {
  AgentDetail,
  PatchedAgentDetail,
  PatchedTaskDetail,
  TaskDetail,
} from '../models/index';
import {
    AgentDetailFromJSON,
    AgentDetailToJSON,
    PatchedAgentDetailFromJSON,
    PatchedAgentDetailToJSON,
    PatchedTaskDetailFromJSON,
    PatchedTaskDetailToJSON,
    TaskDetailFromJSON,
    TaskDetailToJSON,
} from '../models/index';

export interface AgentsBrowserUsePartialUpdateRequest {
    id: string;
    patchedAgentDetail?: Omit<PatchedAgentDetail, 'id'|'user_email'|'created_at'|'updated_at'>;
}

export interface AgentsBrowserUseTasksPartialUpdateRequest {
    agentId: string;
    id: string;
    patchedTaskDetail?: Omit<PatchedTaskDetail, 'id'|'agent_id'|'status'|'created_at'|'updated_at'|'error_message'>;
}

/**
 * 
 */
export class AgentsApi extends runtime.BaseAPI {

    /**
     * ViewSet for managing BrowserUseAgents.
     */
    async agentsBrowserUsePartialUpdateRaw(requestParameters: AgentsBrowserUsePartialUpdateRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<AgentDetail>> {
        if (requestParameters['id'] == null) {
            throw new runtime.RequiredError(
                'id',
                'Required parameter "id" was null or undefined when calling agentsBrowserUsePartialUpdate().'
            );
        }

        const queryParameters: any = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Authorization"] = await this.configuration.apiKey("Authorization"); // tokenAuth authentication
        }

        const response = await this.request({
            path: `/agents/browser-use/{id}/`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters['id']))),
            method: 'PATCH',
            headers: headerParameters,
            query: queryParameters,
            body: PatchedAgentDetailToJSON(requestParameters['patchedAgentDetail']),
        }, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => AgentDetailFromJSON(jsonValue));
    }

    /**
     * ViewSet for managing BrowserUseAgents.
     */
    async agentsBrowserUsePartialUpdate(id: string, patchedAgentDetail?: PatchedAgentDetail, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<AgentDetail> {
        const response = await this.agentsBrowserUsePartialUpdateRaw({ id: id, patchedAgentDetail: patchedAgentDetail }, initOverrides);
        return await response.value();
    }

    /**
     * ViewSet for managing BrowserUseAgentTasks. Supports both agent-specific and user-wide task operations.
     */
    async agentsBrowserUseTasksPartialUpdateRaw(requestParameters: AgentsBrowserUseTasksPartialUpdateRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<TaskDetail>> {
        if (requestParameters['agentId'] == null) {
            throw new runtime.RequiredError(
                'agentId',
                'Required parameter "agentId" was null or undefined when calling agentsBrowserUseTasksPartialUpdate().'
            );
        }

        if (requestParameters['id'] == null) {
            throw new runtime.RequiredError(
                'id',
                'Required parameter "id" was null or undefined when calling agentsBrowserUseTasksPartialUpdate().'
            );
        }

        const queryParameters: any = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Authorization"] = await this.configuration.apiKey("Authorization"); // tokenAuth authentication
        }

        const response = await this.request({
            path: `/agents/browser-use/{agentId}/tasks/{id}/`.replace(`{${"agentId"}}`, encodeURIComponent(String(requestParameters['agentId']))).replace(`{${"id"}}`, encodeURIComponent(String(requestParameters['id']))),
            method: 'PATCH',
            headers: headerParameters,
            query: queryParameters,
            body: PatchedTaskDetailToJSON(requestParameters['patchedTaskDetail']),
        }, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => TaskDetailFromJSON(jsonValue));
    }

    /**
     * ViewSet for managing BrowserUseAgentTasks. Supports both agent-specific and user-wide task operations.
     */
    async agentsBrowserUseTasksPartialUpdate(agentId: string, id: string, patchedTaskDetail?: PatchedTaskDetail, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<TaskDetail> {
        const response = await this.agentsBrowserUseTasksPartialUpdateRaw({ agentId: agentId, id: id, patchedTaskDetail: patchedTaskDetail }, initOverrides);
        return await response.value();
    }

}
