// @gobii-ai/client/src/apis/TasksApi.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Gobii API
 * API for Gobii browser automation platform
 *
 * The version of the OpenAPI document: 0.1.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


import * as runtime from '../runtime';
import type {
  PatchedTaskDetail,
  TaskDetail,
} from '../models/index';
import {
    PatchedTaskDetailFromJSON,
    PatchedTaskDetailToJSON,
    TaskDetailFromJSON,
    TaskDetailToJSON,
} from '../models/index';

export interface TasksBrowserUsePartialUpdateRequest {
    id: string;
    patchedTaskDetail?: Omit<PatchedTaskDetail, 'id'|'agent_id'|'status'|'created_at'|'updated_at'|'error_message'>;
}

/**
 * 
 */
export class TasksApi extends runtime.BaseAPI {

    /**
     * ViewSet for managing BrowserUseAgentTasks. Supports both agent-specific and user-wide task operations.
     */
    async tasksBrowserUsePartialUpdateRaw(requestParameters: TasksBrowserUsePartialUpdateRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<TaskDetail>> {
        if (requestParameters['id'] == null) {
            throw new runtime.RequiredError(
                'id',
                'Required parameter "id" was null or undefined when calling tasksBrowserUsePartialUpdate().'
            );
        }

        const queryParameters: any = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Authorization"] = await this.configuration.apiKey("Authorization"); // tokenAuth authentication
        }

        const response = await this.request({
            path: `/tasks/browser-use/{id}/`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters['id']))),
            method: 'PATCH',
            headers: headerParameters,
            query: queryParameters,
            body: PatchedTaskDetailToJSON(requestParameters['patchedTaskDetail']),
        }, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => TaskDetailFromJSON(jsonValue));
    }

    /**
     * ViewSet for managing BrowserUseAgentTasks. Supports both agent-specific and user-wide task operations.
     */
    async tasksBrowserUsePartialUpdate(id: string, patchedTaskDetail?: PatchedTaskDetail, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<TaskDetail> {
        const response = await this.tasksBrowserUsePartialUpdateRaw({ id: id, patchedTaskDetail: patchedTaskDetail }, initOverrides);
        return await response.value();
    }

}
